File: 2-Android With Java/Day4/Day2_Lab/app/src/main/java/com/example/day2_lab/MainActivity.java
package com.example.day2_lab;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

  private static final String TAG="MainActivity" ;
    private Button btnNext;
    private Button btnExit;

    private EditText edtPhone;

    private EditText edtText;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_main);
        Log.i(TAG, "onCreate: ");

        //btnExit=findViewById(R.id.btnExit);
        btnNext = findViewById(R.id.btnNext);
        edtPhone = findViewById(R.id.edtPhone);

        edtText = findViewById(R.id.edtText);

        btnNext.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this , SecondActivity.class);
                intent.putExtra("Mobile_Number" ,edtPhone.getText().toString() );
                intent.putExtra("Message" , edtText.getText().toString());
                startActivity(intent);
            }
        });




    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.i(TAG, "onStart: ");
    }

    @Override
    protected void onRestart() {
        super.onRestart();
        Log.i(TAG, "onRestart: ");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.i(TAG, "onResume: ");
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.i(TAG, "onPause: ");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.d(TAG, "onStop: ");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.i(TAG, "onDestroy: ");
    }

    public void exit(View view) {
        finish();
    }
}

File: 4-Android with Kotlin/Day3/BoundService/app/src/main/java/com/example/boundservice/MainActivity.kt
package com.example.boundservice

import android.content.ComponentName
import android.content.Context
import android.content.Intent
import android.content.ServiceConnection
import android.os.Bundle
import android.os.IBinder
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import javax.security.auth.login.LoginException

class MainActivity : AppCompatActivity()
{
    lateinit var myService: BoundService
    var isBound :Boolean = false
    lateinit var tvTime : TextView
    lateinit var btnShow : Button
    lateinit var btnDownload : Button
    lateinit var tvurl : EditText


    private var myConnection:ServiceConnection = object : ServiceConnection {
        override fun onServiceConnected(name: ComponentName?, service: IBinder?) {
            val binder :BoundService.MyLocalBinder = service as BoundService.MyLocalBinder
            myService=binder.getService()
            isBound = true
        }

        override fun onServiceDisconnected(name: ComponentName?) {
            isBound = false
        }
    }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val intent = Intent(this , BoundService::class.java)
        bindService(intent , myConnection , Context.BIND_AUTO_CREATE)

        tvTime = findViewById(R.id.textView)
        btnShow = findViewById(R.id.button)
        btnDownload = findViewById(R.id.download)
        tvurl = findViewById(R.id.url)
        btnShow.setOnClickListener{
            showTime()
        }
        btnDownload.setOnClickListener{
            var intent = Intent(this@MainActivity ,DownloadService::class.java )
            intent.putExtra("url" ,tvurl.text.toString() )
            startForegroundService(intent)

        }



    }
    fun showTime()
    {
        val currentTime :String=myService.getCurrentTime()
        Log.i("TAG", "showTime: $currentTime ")
        tvTime.text=currentTime

    }
}